internal/fwserver/server_planresourcechange.go

```diff
@@ -250,6 +251,16 @@ func (s *Server) PlanResourceChange(ctx context.Context, req *PlanResourceChange
 		}
 	}
 
+	// Skip resource-level ModifyPlan for automatic deferrals
+	// unless ProviderDeferredBehavior.EnablePlanModification is true
+	if s.deferred != nil && !req.ResourceBehavior.ProviderDeferred.EnablePlanModification {
+		logging.FrameworkDebug(ctx, "Provider has deferred response configured, automatically returning deferred response.")
+		resp.Deferred = &resource.Deferred{
+			Reason: resource.DeferredReason(s.deferred.Reason),
+		}
+		return
+	}
```

Comment:
I'm interested what the potential impact might be with introducing the automatic deferral after all of these have run:
- Resource `Configure` method
- Provider-defined schema default plan modifiers
- Null computed attributes marked as unknown
- Provider-defined schema plan modifiers

I think this should be okay since we don't explicitly pass API client information here, but some of our documentation refers to plan modification as "online", like the blurb about validation: https://developer.hashicorp.com/terraform/plugin/framework/resources/validate-configuration

If we do go with this route, I think we should update the naming/documentation of `ResourceBehavior.EnablePlanModification` to be more explicit about "what" plan modification logic is being enabled/disabled. Might need to bikeshed a name like `EnableModifyPlan`, `EnableResourcePlanModification`, etc 🤔 

Any thoughts on this @bflad ?

Reply:
That is a fair point. Another option is doing similar as the RPCs and returning without doing _anything_ and would technically be the safest choice. I had hoped that since most of the list up there shouldn't be doing anything with using APIs, etc. we could improve the plan output for future rounds, but that could be changed later as an enhancement, rather than the reverse of taking functionality away being a breaking change. 

I guess I'll concede we should be safer here and return super early before anything runs. We can always change that decision later. I do worry about the null to unknown logic not running although maybe core will already do something like that in the plan output based on the configuration (something also worth checking with ReadDataSource).

Reply:
For safe routes, would it be a bad idea to just duplicate the "marked computed nils as unknown" logic after checking for a deferral at the top of `PlanResourceChange`?

```go
modifiedPlan, err := tftypes.Transform(resp.PlannedState.Raw, MarkComputedNilsAsUnknown(ctx, req.Config.Raw, req.ResourceSchema))
```

I'm also cool with just coming back later and enhancing it once we have more info, using `ProposedNewState` like SDKv2: https://github.com/hashicorp/terraform-plugin-sdk/blob/5fee97c1b78c352a1195194a12a9ad514508e3c6/helper/schema/grpc_provider.go#L777-L792

Reply:
Probably fine but I guess it might be good to evaluate the "current" behavior if the provider side just reflects that data without adding the extra logic as maybe its not needed. That evaluation does not need to happen immediately though, which is nice.

Reply:
I've added a flag in [31f33f8](https://github.com/hashicorp/terraform-plugin-framework/pull/1002/commits/31f33f83f969b60cf83a3194165ce66965fa8491) to essentially skip everything except "Null computed attributes marked as unknown" logic. Unfortunately, there's a lot of prerequisites to calling `MarkComputedNilsAsUnknown()` that I felt was bloating the deferral check block, but if the flag logic is too complicated, I can go back to duplicating the logic in the deferral check.

Reply:
My only reservation in adding a flag to all the other pieces of plan logic is just future maintenance/changes to `PlanResourceChange`, i.e. it's easier to describe just the operations that automatic deferred plans needs to run, rather than describe all the operations (and future operations) that shouldn't run.

If there are a ton of operations and such that need to run for the computed marking, maybe we should just return early with `ProposedNewState` (like [SDKv2](https://github.com/hashicorp/terraform-plugin-sdk/blob/5fee97c1b78c352a1195194a12a9ad514508e3c6/helper/schema/grpc_provider.go#L777-L792) is doing) for now and treat it as an enhancement once deferred actions/stacks is more in-use.

Plans with deferred actions aren't held to the consistency rules by Terraform between rounds, so this is mostly a UX problem